### PR TITLE
Update minimal version of hpp-fcl to 2.0.0

### PR DESCRIPTION
**Pull request by jcarpent**
Solve #2080

### CODE REVIEW

**Comment by singingwolfboy** [comment]
Could you also remove the `from distutils.version import LooseVersion` line? This project should not import or depend on `distutils`, since it has been removed from Python 3.12.

**Comment by jcarpent** [comment]
Yes, sure. This is done in [7b36e32](https://github.com/stack-of-tasks/pinocchio/pull/2086/commits/7b36e323ad3434daadf2ab7a9983deace6782a2b)

File: unittest/urdf.cpp [hunk 1]
```diff
@@ -56,18 +56,11 @@ BOOST_AUTO_TEST_CASE ( build_model_simple_humanoid )
...
-                                     HPP_FCL_PATCH_VERSION>3))))
-#  define PINOCCHIO_HPP_FCL_SUPERIOR_TO_1_1_3
-#endif
-
-#if defined(PINOCCHIO_HPP_FCL_SUPERIOR_TO_1_1_3) && !defined(PINOCCHIO_URDFDOM_COLLISION_WITH_GROUP_NAME)
+#ifndef PINOCCHIO_URDFDOM_COLLISION_WITH_GROUP_NAME
   BOOST_CHECK_EQUAL(geomModel.geometryObjects[1].geometry->getNodeType(), hpp::fcl::GEOM_CONVEX);
-#undef PINOCCHIO_HPP_FCL_SUPERIOR_TO_1_1_3
-#else // PINOCCHIO_HPP_FCL_SUPERIOR_TO_1_1_3 && !PINOCCHIO_URDFDOM_COLLISION_WITH_GROUP_NAME
+#else PINOCCHIO_URDFDOM_COLLISION_WITH_GROUP_NAME
```
Comment (jorisv) [inline unittest/urdf.cpp:61]:
```suggestion
#else // PINOCCHIO_URDFDOM_COLLISION_WITH_GROUP_NAME
```

File: include/pinocchio/multibody/fcl.hpp [hunk 2]
```diff
@@ -47,30 +47,15 @@
...
-#include <boost/make_shared.hpp>
-#endif
 
 #include <boost/foreach.hpp>
 
 namespace pinocchio
 {
-#ifdef PINOCCHIO_HPPFCL_USE_STD_SHARED_PTR
   using std::shared_ptr;
   using std::make_shared;
```
[the remark this answers is not in the document]
Comment (jcarpent) [reply include/pinocchio/multibody/fcl.hpp:57]: Do agree. We can change this practice for Pinocchio 3x.

[the remark this answers is not in the document]
Comment (jorisv) [reply include/pinocchio/multibody/fcl.hpp:57]: Ok, the issue is created

**Pull request merged**